src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -517,13 +517,14 @@ private HttpClient CreateHttpClient()
                 {
                     httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
                 }
-                if (_httpConnectionOptions.Cookies != null)
+
+                // Only access HttpClientHandler.ClientCertificates and HttpClientHandler.CookieContainer
+                // if the user has configured those options
+                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException
+                if (_httpConnectionOptions?.Cookies.Count > 0)
```

Comment:
Based on the old code it looks like `_httpConnectionOptions` is never null. Was this meant to be the following?

```suggestion
                if (_httpConnectionOptions.Cookies?.Count > 0)
```